src/app.js

```diff
@@ -72,78 +59,9 @@ const create = ({ templatesDir, db, logger }) => {
         app.use(createAuthMiddleware(config.userIdHeader, config.oryKetoReadUrl));
     }
 
-    const templates = readTemplates(templatesDir);
-    const reportHandlers = createReportHandlers(templates);
-    const reportHandlersWithSuffixes = Object.fromEntries(
-        Object.entries(reportHandlers)
-            .reduce((pv, [key, val]) => [
-                ...pv,
-                [`${key}.csv`, val],
-                [`${key}.xlsx`, val],
-                [`${key}.html`, val],
-            ], []),
-    );
-    const routeHandlers = { ...handlerMap, ...reportHandlersWithSuffixes };
-    logger.push({ routes: Object.keys(routeHandlers) }).log('Serving routes');
-    app.use(router(routeHandlers));
-
-    // Serialise the body to the type we're interested in
-    app.use(async (ctx, next) => {
-        const suffix = ctx.request.path.split('.').pop();
-        switch (suffix) {
-            case 'xlsx': {
-                ctx.state.logger.log('Setting XLSX response');
-
-                const reportName = ctx.request.path.substr(ctx.request.path.lastIndexOf('/'), ctx.request.path.length)
-                    .replace('/', '').replace('.xlsx', '');
-                const conversion = conversionFactory({
-                    extract: async ({ html, ...restOptions }) => {
-                        const tmpHtmlPath = `/tmp/${reportName}_${Date.now()}.html`;
-
-                        await fs.writeFile(tmpHtmlPath, html);
-
-                        const result = await chromeEval({
-                            ...restOptions,
-                            html: tmpHtmlPath,
-                            scriptFn: conversionFactory.getScriptFn(),
-                        });
+    startOperator(reportData);
```

Comment:
Does the whole service function if an error occurs starting the operator? It looks like this function will never throw. Should the service crash? Should there be some retry logic?

Reply:
If you decide on retry logic; I recommend async-retry. Example here: https://github.com/modusbox/connection-manager-api/blob/873ed4b86e8592b2234d02f2300e9a0e736f168f/server/src/db/database.js#L63

But I think there shouldn't really be a problem crashing the service if this function fails.

Reply:
Detached by later commits. New comment: https://github.com/mojaloop/reporting/pull/30/files#r777928113